packages/client/hmi-client/src/workflow/ops/stratify-mira/tera-stratify-mira.vue

```diff
@@ -208,16 +234,89 @@ const inputChangeHandler = async () => {
 	}
 };
 
-const saveNewModel = async () => {
-	if (!model.value || !newModelName.value) return;
-	model.value.header.name = newModelName.value;
+const saveNewModel = async (modelName: string, options: Record<string, boolean>) => {
+	if (!model.value || !modelName) return;
+	model.value.header.name = modelName;
 
 	const projectResource = useProjects();
 	const modelData = await createModel(model.value);
 	const projectId = projectResource.activeProject.value?.id;
 
 	if (!modelData) return;
-	await projectResource.addAsset(AssetType.Models, modelData.id, projectId);
+
+	if (options.addToProject) {
+		await projectResource.addAsset(AssetType.Models, modelData.id, projectId);
+	}
+
+	if (options.appendOutputPort) {
+		await appendOutputPort(modelData.id);
+	}
+};
+
+const appendOutputPort = async (modelId: string) => {
+	// fetch the model that was just created
+	const newModel = await getModel(modelId);
+	if (!newModel) return;
+
+	// set default configuration for the newly created model
+	await addDefaultConfiguration(newModel);
+
+	// setting timeout...elastic search might not update default config in time
+	setTimeout(async () => {
+		const configurationList = await getModelConfigurations(newModel.id);
+		configurationList.forEach((configuration) => {
+			emit('append-output-port', {
+				id: uuidv4(),
+				label: `${newModel.header.name} - ${configuration.name}`,
+				type: 'modelConfigId',
+				value: [configuration.id]
+			});
+		});
+	}, 800);
+};
+
+const initialize = (editorInstance) => {
+	editor.value = editorInstance;
+};
+
+const runCodeStratify = () => {
+	const code = editor.value?.getValue();
+	if (!code) return;
+
+	resetModel();
+
+	const messageContent = {
+		silent: false,
+		store_history: false,
+		user_expressions: {},
+		allow_stdin: true,
+		stop_on_error: false,
+		code
+	};
+
+	let executedCode = '';
+
+	kernelManager
+		.sendMessage('execute_request', messageContent)
+		?.register('execute_input', (data) => {
+			executedCode = data.content.code;
+		})
+		?.register('stream', (data) => {
+			console.log('stream', data);
+		})
+		?.register('error', (data) => {
+			logger.error(`${data.content.ename}: ${data.content.evalue}`);
+		})
+		?.register('model_preview', (data) => {
+			handleModelPreview(data);
+
+			if (executedCode) {
+				const state = _.cloneDeep(props.node.state);
+				state.hasCodeBeenRun = true;
```

Comment:
This is fine, but you need to do the reverse, when you stratify via the wizard, you need to set `hasCodeBeenRun` to false.

Reply:
done